pkg/env/readers.go

```diff
@@ -71,16 +109,16 @@ func testReader(fallback cfg.Reader, overrides map[string]interface{}) cfg.Reade
 //
 // The provided value is serialized to yaml and so can be structured data.
 func FakeTestConfig(fName string, ptr interface{}) func() {
-	if _, ok := testOverrides[fName]; ok {
+	if _, ok := overrides.load(fName); ok {
 		// This is not ideal.  We would prefer to return an error.
 		// However, this function's signature does not support it and we
 		// don't want to incur the backwards-incompatibility of changing
 		// it right now.
 		panic(fmt.Errorf("repeated test override of '%s'", fName))
 	}
 
-	testOverrides[fName] = ptr
+	overrides.add(fName, ptr)
```

Comment:
This still has a race condition, because you're not copying things. You need a per-goroutine _copy_ of the structure, else you're still sharing memory.